src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/PrintRequestStatus.kt

```diff
@@ -0,0 +1,70 @@
+package uk.gov.dluhc.printapi.rds.entity
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.GenericGenerator
+import org.hibernate.annotations.Type
+import org.hibernate.annotations.UpdateTimestamp
+import org.springframework.data.annotation.LastModifiedBy
+import org.springframework.data.jpa.domain.support.AuditingEntityListener
+import uk.gov.dluhc.printapi.database.entity.Status
+import uk.gov.dluhc.printapi.rds.repository.UUIDCharType
+import uk.gov.dluhc.printapi.rds.repository.UseExistingOrGenerateUUID
+import java.time.Instant
+import java.util.UUID
+import javax.persistence.Entity
+import javax.persistence.EntityListeners
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.Table
+import javax.persistence.Version
+import javax.validation.constraints.NotNull
+import javax.validation.constraints.Size
+
+@Table
+@Entity
+@EntityListeners(AuditingEntityListener::class)
+class PrintRequestStatus(
+    @Id
+    @Type(type = UUIDCharType)
+    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = UseExistingOrGenerateUUID.NAME)
+    var id: UUID? = null,
+
+    @NotNull
+    @Size(max = 255)
+    var status: Status? = null,
+
+    @NotNull
+    var eventDateTime: Instant? = null, // either the "timestamp" from the print provider, or the current time
```

Comment:
`eventDateTime` is marked as non nullable. However Initially this `eventDateTime` could be null but I remember @matt-wills-valtech mentioned that we will set this to same time as that of dateCreated when initial record is created for print request status

Reply:
It should never be null - as you said it will be set to the same time as dateCreated when an initial record is created.